From 236552ff5b9f1ebf666d8d0e9850007dcce03d26 Mon Sep 17 00:00:00 2001
From: Serhii Popovych <spopovyc@cisco.com>
Date: Wed, 10 Feb 2016 16:32:44 +0000
Subject: [PATCH] perl: Replace -w option in shebangs with modern "use
 warnings"

In some builds we might provide ac_cv_path_PERL as /usr/bin/env perl
to use newer version of the perl from users PATH rather than
older from standard system path.

However using /usr/bin/env perl -w from shebang line isn't
possible because it translates to something like
/usr/bin/env -w perl and env complains about illegal option.

To address this we can remove -w option from perl shebang
line and add "use warnings" statement.

Upstream-Status: Pending
Signed-off-by: Serhii Popovych <spopovyc@cisco.com>
---
 bin/autom4te.in                                          | 3 ++-
 bin/autoreconf.in                                        | 3 ++-
 bin/autoscan.in                                          | 3 ++-
 bin/autoupdate.in                                        | 3 ++-
 bin/ifnames.in                                           | 3 ++-
 5 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/bin/autom4te.in b/bin/autom4te.in
index 11773c9..a8f5e41 100644
--- a/bin/autom4te.in
+++ b/bin/autom4te.in
@@ -1,4 +1,4 @@
-#! @PERL@ -w
+#! @PERL@
 # -*- perl -*-
 # @configure_input@
 
@@ -42,6 +42,7 @@ use Autom4te::General;
 use Autom4te::XFile;
 use File::Basename;
 use strict;
+use warnings;
 
 # Data directory.
 my $pkgdatadir = $ENV{'AC_MACRODIR'} || '@pkgdatadir@';
diff --git a/bin/autoreconf.in b/bin/autoreconf.in
index e245db4..1a318cb 100644
--- a/bin/autoreconf.in
+++ b/bin/autoreconf.in
@@ -1,4 +1,4 @@
-#! @PERL@ -w
+#! @PERL@
 # -*- perl -*-
 # @configure_input@
 
@@ -45,6 +45,7 @@ use Autom4te::XFile;
 # Do not use Cwd::chdir, since it might hang.
 use Cwd 'cwd';
 use strict;
+use warnings;
 
 ## ----------- ##
 ## Variables.  ##
diff --git a/bin/autoscan.in b/bin/autoscan.in
index a67c48d..b931249 100644
--- a/bin/autoscan.in
+++ b/bin/autoscan.in
@@ -1,4 +1,4 @@
-#! @PERL@ -w
+#! @PERL@
 # -*- perl -*-
 # @configure_input@
 
@@ -43,6 +43,7 @@ use Autom4te::XFile;
 use File::Basename;
 use File::Find;
 use strict;
+use warnings;
 
 use vars qw(@cfiles @makefiles @shfiles @subdirs %printed);
 
diff --git a/bin/autoupdate.in b/bin/autoupdate.in
index 9737d49..92cb147 100644
--- a/bin/autoupdate.in
+++ b/bin/autoupdate.in
@@ -1,4 +1,4 @@
-#! @PERL@ -w
+#! @PERL@
 # -*- perl -*-
 # @configure_input@
 
@@ -44,6 +44,7 @@ use Autom4te::General;
 use Autom4te::XFile;
 use File::Basename;
 use strict;
+use warnings;
 
 # Lib files.
 my $autom4te = $ENV{'AUTOM4TE'} || '@bindir@/@autom4te-name@';
diff --git a/bin/ifnames.in b/bin/ifnames.in
index ba2cd05..74b0278 100644
--- a/bin/ifnames.in
+++ b/bin/ifnames.in
@@ -1,4 +1,4 @@
-#! @PERL@ -w
+#! @PERL@
 # -*- perl -*-
 # @configure_input@
 
@@ -44,6 +44,7 @@ BEGIN
 use Autom4te::General;
 use Autom4te::XFile;
 use Autom4te::FileUtils;
+use warnings;
 
 # $HELP
 # -----
-- 
2.3.0

